openTSNE/affinity.py

```diff
@@ -129,14 +133,27 @@ def __init__(
         n_jobs=1,
         random_state=None,
         verbose=False,
+        k_neighbors="auto",
     ):
         self.n_samples = data.shape[0]
+        
+        if k_neighbors=="auto":
+            self.k_neighbors = min(self.n_samples - 1, int(3 * perplexity))
+        else:
+            self.k_neighbors = k_neighbors
```

Comment:
Please change this so `self.k_neighbors` is set to the user-defined parameter. So, if I instantiate `x = Class(k="auto")`, then I'd be surprised if `x.k` is set to `90` for instance. I'd expect it to be set to `"auto"`. This type of interface also breaks if I change `x.k` myself e.g. `x.k = "auto"`. Then, nothing will work anymore.

I realise it may not be as relevant here because it's all going on inside the same function, but it's good practice. I also realise now this is technically not what happens with `self.perplexity`, but at least there's a warning there.

So I would suggest saving `self.k_neighbors = "auto"`, and using a local variable to determine the proper number of neighbors. Then, we'll have to do something similar in `to_new`.

Reply:
Fixed. I had the same issue raised with my recent PR to sklearn. They use `self._k_neighbors` naming scheme for this purpose, so that's what I used here now.